lib/local-cluster/lib/Cardano/Wallet/Launch/Cluster/Monitoring/Monitor.hs

```diff
@@ -98,18 +120,51 @@ timedMonitor initialState = do
 -- | Start a monitoring service, returning a tracer to write `Phase` values to
 -- and a function to interact with the monitoring service
 withMonitoring
-    :: Tracer IO MsgHttpMonitoring
+    :: HasSNetworkId n
+    => SNetworkId n
+    -> NodeConnVar
+    -> Config
+    -> Tracer IO MsgHttpMonitoring
     -- ^ Tracer for logging the monitoring operations
     -> MonitorConfiguration
     -- ^ Configuration for the monitoring service
-    -> ContT () IO (Tracer IO Phase, RunQuery IO)
-withMonitoring tr MonitorConfiguration{..} = do
+    -> ContT () IO (Tracer IO Phase, (RunQuery IO, RunApplicationQ IO))
+withMonitoring network conn clusterConfig tr config = do
+    (port, tracer) <- withMonitoringServer network conn clusterConfig tr config
+    queries <- withMonitoringClient network port tr
+    pure (tracer, queries)
+
+withMonitoringClient
+    :: HasSNetworkId n
+    => SNetworkId n
+    -> PortNumber
+    -> Tracer IO MsgHttpMonitoring
+    -> ContT () IO (RunQuery IO, RunApplicationQ IO)
+withMonitoringClient network port tr = do
+    liftIO $ traceWith tr MsgHttpMonitoringClientStarted
+    queries <- withHttpClient network (MsgHttpMonitoringQuery >$< tr) port
+    ContT $ \k -> do
+        k queries
+        traceWith tr MsgHttpMonitoringClientStopped
+
+withMonitoringServer
+    :: HasSNetworkId n
+    => SNetworkId n
+    -> NodeConnVar
+    -> Config
+    -> Tracer IO MsgHttpMonitoring
+    -> MonitorConfiguration
+    -> ContT () IO (PortNumber, Tracer IO Phase)
+withMonitoringServer network conn clusterConfig tr MonitorConfiguration{..} = do
     monitor <- liftIO $ withTracingState timedMonitor monitorInitialState
     port <- liftIO $ maybe getRandomPort pure monitorPort
     liftIO $ traceWith tr $ MsgHttpMonitoringPort port
-    withHttpServer port $ mkHandlers $ rmap History monitor
+    withHttpServer
+        network
+        port
+        (mkControlHandlers $ rmap History monitor)
+        (mkApplicationHandlers conn clusterConfig)
```

Comment:
But the `mkApplicationHandlers` are no longer part of the `Monitoring` concern proper, are they? 🤔 Maybe move `withMonitoringServer` to the testing code?

Reply:
The HTTP server is the same, so we have to compose them. The api backed by `ApplicationHandlers` are intended to be offered within the local-cluster features.